pkg/3scale/amp/component/deployments_lister.go

```diff
@@ -0,0 +1,52 @@
+package component
+
+type SystemDatabaseType string
+
+const (
+	SystemDatabaseTypeInternalMySQL      SystemDatabaseType = "mysql"
+	SystemDatabaseTypeInternalPostgreSQL SystemDatabaseType = "postgresql"
+	SystemDatabaseTypeExternal           SystemDatabaseType = "external"
+)
+
+type DeploymentsLister struct {
+	SystemDatabaseType     SystemDatabaseType
+	ExternalRedisDatabases bool
+	ExternalZyncDatabase   bool
+}
+
+func (d *DeploymentsLister) DeploymentNames() []string {
+	var deployments []string
+	deployments = append(deployments,
+		ApicastStagingName,
+		ApicastProductionName,
+		BackendListenerName,
+		BackendWorkerName,
+		BackendCronName,
+		SystemMemcachedDeploymentName,
+		SystemAppDeploymentName,
+		SystemSidekiqName,
+		SystemSphinxDeploymentName,
+		ZyncName,
+		ZyncQueDeploymentName,
+	)
+
+	switch d.SystemDatabaseType {
+	case SystemDatabaseTypeInternalMySQL:
+		deployments = append(deployments, SystemMySQLDeploymentName)
+	case SystemDatabaseTypeInternalPostgreSQL:
+		deployments = append(deployments, SystemPostgreSQLDeploymentName)
+	}
+
+	if !d.ExternalRedisDatabases {
+		deployments = append(deployments,
+			BackendRedisDeploymentName,
+			SystemRedisDeploymentName,
+		)
+	}
+
+	if !d.ExternalZyncDatabase {
+		deployments = append(deployments)
```

Comment:
missing something, right?

Reply:
:scream: you are right. Fixed, thanks! :+1: